AutoCollection/CorrelationContextManager.ts

```diff
@@ -35,16 +36,16 @@ export class CorrelationContextManager {
     /**
      *  A helper to generate objects conforming to the CorrelationContext interface
      */
-    public static generateContextObject(parentId?: string, operationName?: string, operationId?: string): CorrelationContext {
-        operationId = operationId || Util.newGuid();
+    public static generateContextObject(operationId: string, parentId?: string, operationName?: string, correlationContextHeader?: string): CorrelationContext {
         parentId = parentId || operationId;
         
         if (this.enabled) {
             return {
                 operation: {
                     name: operationName,
                     id: operationId,
-                    parentId: parentId
+                    parentId: parentId,
+                    correlationContextHeader: correlationContextHeader
```

Comment:
I think we can't deserialize the header into a map, since the spec explicitly says that keys may be duplicated, and we must not aggregate or deduplicate values. I'd prefer to keep this as an opaque string (that we potentially add `, key=value` to the end of), but I'd also be reasonably happy with converting it to an array of `key=value` strings, splitting on `', '` and potentially appending more in future.